hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -176,6 +178,9 @@ public void cleanResources(ConstructorFunction<Object, Throwable> responseCtor,
                 invocation.notifyException(responseCtor.createNew(null));
             }
         }
+        //Try cleaning again later to compensate the Race between `send` and `cleanResources`
```

Comment:
You won't notice it :)

If you do 100k ops/second.. you are generating many times more garbage. Lets say 300k objects per second. If you have 1000 pending invocations, then at most you will generate a few times X garbage (if any garbage is created at all).

We use the same approach on the server side.. scanning over all invocations see if there are problems like timeout. There is no noticeable performance impact.